src/transformers/models/mplugdocowl/image_processing_mplugdocowl.py

```diff
@@ -0,0 +1,515 @@
+            rescale_factor=rescale_factor,
+            do_normalize=do_normalize,
+            image_mean=image_mean,
+            image_std=image_std,
+            do_center_crop=do_center_crop,
+            crop_size=crop_size,
+            do_resize=do_resize,
+            size=size,
+            resample=resample,
+        )
+        # 1. Keep global image to be able to work with it later
+         
+
+        if do_convert_rgb:
+            images = [convert_to_rgb(image) for image in images]
+        patch_images = images.copy()
+        # All transformations expect numpy arrays.
+        images = [to_numpy_array(image) for image in images]
+
+        if input_data_format is None:
+            # We assume that all images have the same channel dimension format.
+            input_data_format = infer_channel_dimension_format(images[0])
+
+        if do_resize:
+            images = [
+                self.resize(image=image, size=size, resample=resample, input_data_format=input_data_format)
+                for image in images
+            ]
+
+        if do_center_crop:
+            images = [
+                self.center_crop(image=image, size=crop_size, input_data_format=input_data_format) for image in images
+            ]
+
+        if do_rescale:
+            images = [
+                self.rescale(image=image, scale=rescale_factor, input_data_format=input_data_format)
+                for image in images
+            ]
+        #breakpoint()
+        if do_shape_adaptive_cropping:
+            patch_images = [self.adaptive_crop(image) for image in patch_images]
+            images.extend(patch_images[0])
+            #breakpoint()
+        if is_scaled_image(images[0]) and do_rescale:
+            logger.warning_once(
+                "It looks like you are trying to rescale already rescaled images. If the input"
+                " images have pixel values between 0 and 1, set `do_rescale=False` to avoid rescaling them again."
+            )
+        if do_normalize:
+            images = [
+                self.normalize(image=image, mean=image_mean, std=image_std, input_data_format=input_data_format)
+                for image in images
+            ]
+        
+        images = [
+            to_channel_dimension_format(image, data_format, input_channel_dim=input_data_format) for image in images
+        ]
+
+            # call the module
+
+        data = {"pixel_values": images}
+        return BatchFeature(data=data, tensor_type=return_tensors)
+
+#image_processor = MPLUGDocOwlImageProcessor()
+#image = Image.open("/home/dana_aubakirova/test_image.tif")
+#pixel_values = image_processor(image, do_rescale=True, do_convert_rgb=True, do_shape_adaptive_cropping=True, #do_resize=True, do_normalize=True, return_tensors=TensorType.PYTORCH,image_mean=(0.48145466, 0.4578275, 0.40821073), image_std=(0.26862954, 0.26130258, 0.27577711),resample=None,size=224)["pixel_values"]
+#breakpoint()
+#print(pixel_values)
```

Comment:
For quality/style checks, before pushing you can run `make fixup` (be sure to install `pip install -e .[quality]` before) and it'll fix missing newlines, spaces, all the linter work, but also docstrings not matching signatures, copies not matching originals, and so on - be sure `ruff` is in your env https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests

Reply:
nice, thanks! will do that:)